### PR TITLE
[Feature] Registration/Login via Shibboleth/SAML

### PR DESCRIPTION
**Pull request by dfuchss**
This PR is the second version (now on local branch for testing) of #2937 

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).
- [ ] Client: I added multiple integration tests (Jest) related to the features (with a high test coverage)
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context
We have determined that for our purposes, integrating Shibboleth for student registration and/or login would be useful. Shibboleth would ensure that only authorized users can log in.  In addition, this would allow us to access student registration numbers directly.

The workflow:
* An user clicks on something like "Login via Shibboleth"
* The user will be redirected to the Shibboleth Login Mask and enters his/her credentials
* The user will be redirected to Artemis and is authenticated
* If this account is a newly created account, an E-Mail containing a password-reset link for the current user will be sent. (The body of the email says something like "please click on the link to get your app password for gitlab / artemis").

Therefore, we can use the SAML/Shibboleth information for account creation / login to artemis and use the integrated user management to create passwords for the connected services (in our case: gitlab)

This PR is related to #2377 

### Description
This feature allows users to Login via SAML2 (Shibboleth) using [Spring Security SAML2 Implementation](https://docs.spring.io/spring-security/site/docs/5.2.8.RELEASE/reference/html/saml2.html). 

- The feature can be completely enabled/disabled via the `saml2` spring profile
- Configuration is dynamic and can be found in `application-saml2.yml`. 
- Multiple IDPs are supported. In this case an auto-generated login page with an IDP selection can be found at /login

To accomplish this a separated security filter chain is added to Artemis. 
Security

- Requests to /saml2/** and /login/** are processed by the new filter chain.
- There is currently an upstream limitation which allows an auto-generated login page only at /login. In the future a custom login page could be used.
- SAML2-authenticated clients can request their JWTToken at `/api/saml2`

Client flow is as follows:
client

### Steps for Testing
- Setup an SAML2 Identity Provider.
  - The configuration for LDAP/docker-compose-artemis would be
```YAML
saml2:
    username-pattern: 'saml2-{first_name}_{last_name}'
    first-name-pattern: '{first_name}'
    last-name-pattern: '{last_name}'
    email-pattern: '{email}'
    registration-number-pattern: '{uid}'
    identity-providers:
        - metadata: http://saml:8080/simplesaml/saml2/idp/metadata.php
          registration-id: testidp
          entity-id: artemis
```
- Enable `saml2` profile
- Configure the IDP and Artemis in application-saml2.yml
- (Reverse proxy: Forward /saml2/** and /login/**)
- Log in with SAML2 (user are created if necessary)

### Test Coverage (Line Coverage)
- UserJWTController: 89%
- SAML2Service: 97%

### Screenshots
Login Page
Login

User data, generated from configuration and SAML2-attributes:

```YAML
username-pattern: 'saml2-{first_name}_{last_name}'
first-name-pattern: '{first_name}'
last-name-pattern: '{last_name}'
email-pattern: '{email}'
registration-number-pattern: '{uid}'
```
User

#### Instructions for local testing (via configured docker compose environment)
See [kit-sdq/Artemis-SAML2-Test-Docker](https://github.com/kit-sdq/Artemis-SAML2-Test-Docker)

### CODE REVIEW

**Comment by dfuchss** [comment]
@LDAP @b-fein this is the new PR.

File: src/main/java/de/tum/in/www1/artemis/domain/User.java [hunk 1]
```diff
@@ -238,7 +238,12 @@ public void setRegistrationNumber(String registrationNumber) {
     }
 
     public String getVisibleRegistrationNumber() {
-        return visibleRegistrationNumberTransient;
+        if (visibleRegistrationNumberTransient != null) {
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/domain/User.java:247]: Please revert this change. It is intended that the registration number is not send with every REST response that includes a user object.
It has to be copied explicitly into the transient field to be sent to the client

Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/domain/User.java:247]: @LDAP resolved.

**Comment by LDAP** [comment]
Setting the local Artemis password does currently fail due to #3041.

**Comment by dfuchss** [comment]
We've tested it on our systems ; now it's time for a review :)

**Comment by dfuchss** [comment]
As always, feel free to ask anything about the PR

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+
+    /**
+     * Constructs a new instance.
+     *
+     * @param properties The SAML2 properties
+     */
+    public SAML2Configuration(final SAML2Properties properties) {
+        this.properties = properties;
+
+        Security.addProvider(new BouncyCastleProvider());
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:52]: have you tested that this does not interfere with the use of SSH keys when the artemis server clones git repositories? We recently had issues here

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    }
+
+    /**
+     * Returns the RelyingPartyRegistrationRepository used by SAML2 configuration.
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:56]: I would appreciate more descriptive explanation here

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.saml2.core.Saml2X509Credential;
+import org.springframework.security.saml2.provider.service.registration.InMemoryRelyingPartyRegistrationRepository;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrations;
+
+/**
+ * This class describes the security configuration for SAML2.
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:32]: I would appreciate a more descriptive explanation here.
Also additional inline comments would help.
If you have reused this from another source (which is completely fine), it would also make sense to reference it here

File: src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java [hunk 5]
```diff
@@ -124,51 +124,51 @@ protected void configure(HttpSecurity http) throws Exception {
         // @formatter:off
         http
             .csrf()
-            .disable()
+                .disable()
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java:127]: please revert changes in this file as they are not necessary for this PR

File: src/main/java/de/tum/in/www1/artemis/service/connectors/SAML2Service.java [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.security.AuthoritiesConstants;
+import de.tum.in.www1.artemis.service.MailService;
+import de.tum.in.www1.artemis.service.user.UserCreationService;
+import de.tum.in.www1.artemis.service.user.UserService;
+import de.tum.in.www1.artemis.web.rest.vm.ManagedUserVM;
+
+/**
+ * This class describes a service for SAML2 authentication.
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/SAML2Service.java:33]: I would appreciate a more descriptive explanation here.
Also additional inline comments would help.

File: src/main/webapp/app/home/home.component.ts [hunk 7]
```diff
@@ -165,6 +181,30 @@ export class HomeComponent implements OnInit, AfterViewChecked {
             .finally(() => (this.isSubmittingLogin = false));
     }
 
+    loginSAML2() {
```
Comment (krusche) [inline src/main/webapp/app/home/home.component.ts:184]: would it make sense to extract this into a separate component?

File: src/main/webapp/app/home/home.component.html [hunk 8]
```diff
@@ -72,24 +72,33 @@ <h1 jhiTranslate="home.title">Welcome to Artemis!</h1>
...
+                        <div class="btn-toolbar">
+                            <button
+                                type="submit"
+                                [disabled]="isSubmittingLogin || !acceptTerms || !password || password.length < 4 || !username || username.length < 4"
+                                class="btn btn-primary btn-space"
+                            >
+                                <span *ngIf="isSubmittingLogin" class="mr-1"><fa-icon [icon]="'circle-notch'" spin="true"></fa-icon></span>
+                                <span jhiTranslate="login.form.button"> Sign in </span>
+                            </button>
+                            <div *ngIf="saml2Enabled">
```
Comment (krusche) [inline src/main/webapp/app/home/home.component.html:84]: would it make sense to extract this into a separate component?

File: src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts [hunk 9]
```diff
@@ -28,4 +28,5 @@ export class ProfileInfo {
     public accountName?: string;
     public versionControlUrl?: string;
     public programmingLanguageFeatures: ProgrammingLanguageFeature[];
+    public saml2?: { ['button-label']: string; ['enable-password']: boolean };
```
Comment (krusche) [inline src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts:31]: Please extract two attributes here and avoid the use of { ... } as type

**Review by krusche — DISMISSED** [review]
In general the code looks good and useful, however I would appreciate a bit more documentation. Also it would be great if we could try to extract some client code into separate components otherwise the home.component would become too complex.

**Important:** Please add some tests for the new code. I hope this is possible with the use of mocks.

File: src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java (hunk 5 above)
Comment (LDAP) [reply src/main/java/de/tum/in/www1/artemis/config/SecurityConfiguration.java]: Done.

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java (hunk 3 above)
Comment (LDAP) [reply src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:59]: I added some comments

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java (hunk 4 above)
Comment (LDAP) [reply src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:32]: I added some comments.
I did come up with this on my own :)

File: src/main/java/de/tum/in/www1/artemis/service/connectors/SAML2Service.java (hunk 6 above)
Comment (LDAP) [reply src/main/java/de/tum/in/www1/artemis/service/connectors/SAML2Service.java:33]: I added some comments

File: src/main/webapp/app/home/home.component.ts (hunk 7 above)
Comment (LDAP) [reply src/main/webapp/app/home/home.component.ts:184]: I am not that familiar with Angular, in this case I would need and appreciate some help.

File: src/main/webapp/app/home/home.component.html (hunk 8 above)
Comment (LDAP) [reply src/main/webapp/app/home/home.component.html:84]: See https://github.com/ls1intum/Artemis/pull/3002#discussion_r602551008

File: src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts (hunk 9 above)
Comment (LDAP) [reply src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts:31]: Can this be achieved without changing the properties on the server side? The server does currently sends an object
```json
{
"saml2": { "button-label": "...", "enable-password": false }
}
```
Bundling related information would be preferable in my opinion.

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java (hunk 2 above)
Comment (dfuchss) [reply src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:55]: Currently, we do not use ssh clone. Could you explain possible issues in more details?

**Comment by dfuchss** [comment]
@krusche ad tests: I think it's quite difficult to create tests for the whole SAML protocol. Especially, it might be helpful to get some information about your testing environment.

File: src/main/java/de/tum/in/www1/artemis/service/connectors/SAML2Service.java [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+ * 
+ * The main method is {@link #handleAuthentication(Saml2AuthenticatedPrincipal)}. The service extracts the user information
+ * from the {@link Saml2AuthenticatedPrincipal} and creates the user, it it does not exist already.
```
Comment (ivanchimeno) [inline src/main/java/de/tum/in/www1/artemis/service/connectors/SAML2Service.java:36]:
```suggestion
 * from the {@link Saml2AuthenticatedPrincipal} and creates the user, if it does not exist already. 
```

File: src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts (hunk 9 above)
Comment (ivanchimeno) [reply src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts:31]: What you can do is create a new type for example in `login.service.ts` and export it:

```ts
export type Saml2 = { ['button-label']: string; ['enable-password']: boolean };
```

and then you can use that type instead of writing out the properties:

```ts
public saml2?: Saml2;
```

**Review by ivanchimeno — COMMENTED** [review]
I tested the branch on TS1 and TS2 and the changes do not seem to break anything on the systems :) Code also looks good from my eyes but since I don't have much experience with SAML2 I can't really give a good review.

File: src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts (hunk 9 above)
Comment (krusche) [reply src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts]: We can convert what the server sends. This is more about the use of actual classes. So instead of using { ['button-label']: string; ['enable-password']: boolean } I propose to use something like

```
export class Saml2Config {
    public string buttonLabel;
    public boolean enablePassowrd;
    constructor(...) {}
}
```

Normally Angular http client can automatically transfer the json into such an object

File: src/main/webapp/app/home/home.component.ts (hunk 7 above)
Comment (krusche) [reply src/main/webapp/app/home/home.component.ts:184]: @ivanchimeno could you help out here? You are more familiar with Angular. I guess it makes sense to have multiple components in the home.component. The parent component (home.component) would then include sub components for login, registration or saml2 based on the configuration received from the server. This means we could split up the code nicely and avoid issues

File: src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java (hunk 2 above)
Comment (krusche) [reply src/main/java/de/tum/in/www1/artemis/config/SAML2Configuration.java:55]: The SSH library in JGit uses custom security providers. I think they try to use the first provider in Security, but I am not fully sure. In case the SSH library uses a different provider, it might fail for various reasons.

Could you activate SSH in one of your test systems to find out if it interferes with this SAML2 Configuration?

**Comment by dfuchss** [comment]
Regarding Tests: I think the only REST endpoint we've created manually is UserJWTController#authorizeSAML2 . The remaining endpoints are generated by the SAML2 Library of Spring Boot by Auto Config. Furthermore, I'm not quite sure about what to test regarding this single method.

**Comment by krusche** [comment]
> Regarding Tests: I think the only REST endpoint we've created manually is UserJWTController#authorizeSAML2 . The remaining endpoints are generated by the SAML2 Library of Spring Boot by Auto Config. Furthermore, I'm not quite sure about what to test regarding this single method.

Invoke the method over the request util in the test classes and verify that the response is correct.
Also make sure that all methods in `SAML2Service.java` are tested.
The first step would be that the code is invoked (this way you make sure that no exceptions are thrown).
The second step would be that you test whether method return values or database content has changed accordingly after invoking those methods. 

Example: if you invoke `createUser` as part of your test, you can use the `userRepository` to check that this user has actually been stored in the database.

In any case you need a new abstract test class because you are using a new profile.

**Comment by dfuchss** [comment]
@ivanchimeno Stephan said that you are familiar with tests in Jenkins/Gitlab context.
I've added the two (most important) test cases for saml to this branch. Acutally the build fails with 

```
de.tum.in.www1.artemis.service.connectors.gitlab.GitLabException: Unable to get ID for user student1
	at de.tum.in.www1.artemis.service.connectors.gitlab.GitLabUserManagementService.getUserIdCreateIfNotExists(GitLabUserManagementService.java:279)
	at de.tum.in.www1.artemis.service.connectors.gitlab.GitLabUserManagementService.createVcsUser(GitLabUserManagementService.java:56)
	at de.tum.in.www1.artemis.service.user.UserCreationService.lambda$createUser$0(UserCreationService.java:161)
	at java.base/java.util.Optional.ifPresent(Optional.java:176)
	at de.tum.in.www1.artemis.service.user.UserCreationService.createUser(UserCreationService.java:161)
[...]
```

Could you please have a look at it? I think you could find my mistake faster than me :)

**Comment by LDAP** [comment]
## Instructions for local testing

1. Clone the repository https://github.com/LDAP/docker-compose-artemis
2. Follow the README and the usual steps to set up Artemis with GitLab and Jenkins
3. Setup SAML using the configuration file and enable the profile in the `.env` file
The env file could look like that:
```
ARTEMIS_DATA_DIR=~/test/.artemis-data
ARTEMIS_CONFIG_DIR=~/test/.artemis
# Artemis tag or branch
ARTEMIS_VERSION=feature/saml
ARTEMIS_SPRING_PROFILES=prod,jenkins,gitlab,artemis,scheduling,saml2

GITLAB_DATA_DIR=~/test/.gitlab/data
GITLAB_CONFIG_DIR=~/test/.gitlab/config
GITLAB_LOG_DIR=~/test/.gitlab/logs

JENKINS_DATA_DIR=~/test/.jenkins

MYSQL_DATA_DIR=~/test/.db
MYSQL_USER=root
MYSQL_ROOT_PASSWORD=test

CUSTOM_FAVICON=./placeholders/favicon.ico
CUSTOM_LOGO=./placeholders/logo.png
CUSTOM_PRIVACY_STATEMENT=./placeholders/privacy.html

ARTEMIS_SERVER_PORT=8080
ARTEMIS_CLIENT_PORT=80
MYSQL_PORT=3306
GITLAB_HTTP_PORT=8880
GITLAB_HTTPS_PORT=8443
GITLAB_SSH_PORT=2222
JENKINS_HTTP_PORT=7780

# Optional and for docker-compose.dev.yml only
# Path to artemis git repository
ARTEMIS_DIR=../Artemis
ARTEMIS_JAR_NAME=Artemis-4.11.0.jar
```
The application-saml2.yml must look like that:
```yaml
saml2:
    username-pattern: 'saml2-{first_name}_{last_name}'
    first-name-pattern: '{first_name}'
    last-name-pattern: '{last_name}'
    email-pattern: '{email}'
    registration-number-pattern: '{uid}'
    identity-providers:
        - metadata: http://saml:8080/simplesaml/saml2/idp/metadata.php
          registration-id: testidp
          entity-id: artemis
          cert-file: # data/saml/cert (optional) Set this path to the Certificate for encryption/signing or leave it blank
          key-file: # data/saml/key path-to-key (optional) Set this path to the Key for encryption/ssigning or leave it blank

info.saml2.button-label: 'SAML Login'
info.saml2.enable-password: true
```

4. Enable Debug output
5. Start the stack using ` docker-compose -f docker-compose.yml -f docker-compose.dev.yml down`
6. Log in with user1 and user1pass
7. User mail addresses are fixed, testing is in this configuration only possible when overwriting the user information in the SAML docker.

**Comment by dfuchss** [comment]
@krusche : @ivanchimeno helped me with the test :)

**Comment by LDAP** [comment]
I am currently fixing the client test.

File: src/main/webapp/app/home/home.component.ts (hunk 7 above)
Comment (ivanchimeno) [reply src/main/webapp/app/home/home.component.ts]: I extracted the saml2 login into its own component, but sadly I can't test it because I didn't set up the testing environment for saml2. @LDAP  could you let me know if the changes didn't break anything?

**Comment by LDAP** [comment]
@ivanchimeno Login does work. But a minor fix was necessary to remove the "Registriere dich" button again ;)
We want students to be able to reset their passwords, but we do not want students to be able to register themselves.

grafik

File: src/main/webapp/app/home/home.component.html [hunk 11]
```diff
@@ -72,24 +72,29 @@ <h1 jhiTranslate="home.title">Welcome to Artemis!</h1>
...
                 </div>
             </div>
         </div>
     </div>
     <div *ngIf="isRegistrationEnabled">
         <br /><br />
-        <div class="d-flex justify-content-center">
-            <span jhiTranslate="global.messages.info.register.noaccount">You don't have an account yet?</span>&nbsp;
-            <a class="alert-link" routerLink="account/register" jhiTranslate="global.messages.info.register.link">Register a new account</a>
+        <div *ngIf="isRegistrationEnabled">
```
Comment (krusche) [inline src/main/webapp/app/home/home.component.html:93]: this if statement is duplicated

File: src/main/webapp/app/core/login/login.service.ts [hunk 12]
```diff
@@ -9,6 +9,8 @@ import { AccountService } from 'app/core/auth/account.service';
 import { JhiWebsocketService } from 'app/core/websocket/websocket.service';
 import { NotificationService } from 'app/shared/notification/notification.service';
 
+export type Saml2 = { ['button-label']: string; ['enable-password']: boolean };
```
Comment (krusche) [inline src/main/webapp/app/core/login/login.service.ts:12]: This is still no class. Please use a proper class in its own file and call if Saml2Config

File: src/main/webapp/app/core/login/login.service.ts [hunk 13]
```diff
@@ -47,6 +49,30 @@ export class LoginService {
         });
     }
 
+    /**
+     * Login the user with SAML2.
+     * @param callback The callback function to use (optional)
+     */
+    loginSAML2(rememberMe: boolean, callback?: any) {
```
Comment (krusche) [inline src/main/webapp/app/core/login/login.service.ts:56]: can we avoid the type `any` here and directly add a proper type? Maybe something like a Promise?

File: src/test/java/de/tum/in/www1/artemis/authentication/UserSaml2IntegrationTest.java [hunk 14]
```diff
@@ -0,0 +1,184 @@
...
+    @AfterEach
+    public void clearAuth() {
+        TestSecurityContextHolder.clearContext();
+        this.database.resetDatabase();
+    }
+
+    private Saml2AuthenticatedPrincipal createPrincipal(String username) {
+        Saml2AuthenticatedPrincipal principal = new DefaultSaml2AuthenticatedPrincipal(username, new HashMap<>() {
+
+            {
```
Comment (krusche) [inline src/test/java/de/tum/in/www1/artemis/authentication/UserSaml2IntegrationTest.java:175]: I guess you could use something like Map.of(...) here to avoid the Static Code Analysis Warning here

File: src/main/webapp/app/core/login/login.service.ts (hunk 13 above)
Comment (ivanchimeno) [reply src/main/webapp/app/core/login/login.service.ts]: I ended up removing the callback since it isn't used

Comment (dfuchss) [reply src/main/webapp/app/core/login/login.service.ts]: @ivanchimeno so it's resolved?

File: src/test/java/de/tum/in/www1/artemis/authentication/UserSaml2IntegrationTest.java (hunk 14 above)
Comment (dfuchss) [reply src/test/java/de/tum/in/www1/artemis/authentication/UserSaml2IntegrationTest.java:175]: I'll take a look

Comment (dfuchss) [reply src/test/java/de/tum/in/www1/artemis/authentication/UserSaml2IntegrationTest.java]: Should be resolved now. I'll test all changes on our test servers.

**Comment by dfuchss** [comment]
The PR is nearly ready for the (maybe) last review.
Currently some properties have no effect to the UI, but I think this might be fixed soon.

Tests have been refactored.

**Comment by ivanchimeno** [comment]
> The PR is nearly ready for the (maybe) last review.
> Currently some properties have no effect to the UI, but I think this might be fixed soon.
> 
> Tests have been refactored.

I pushed some changes that fix the problem.

**Comment by dfuchss** [comment]
@krusche I think the issues are fixed now :)

File: src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts (hunk 9 above)
Comment (dfuchss) [reply src/main/webapp/app/shared/layouts/profiles/profile-info.model.ts]: @ivanchimeno fixed this

**Review by ivanchimeno — APPROVED** [review]
I'm approving because the code looks great and the feature has been tested successfully during the testing session with @dfuchss  and @LDAP . I also deployed it on TS1 and TS2 and didn't find any issues.

Nice job! 👍
